### ISSUE TITLE
need a way to create roles/SA in remote clusters

### ISSUE THREAD

**Issue by jmazzitelli** [kiali/kiali#5987]
We need a way for users to create roles/services accounts on remote clusters. We have something that kinda does it , but its a hack script and I believe this probably isn't going to be good enough: https://github.com/kiali/kiali/blob/master/hack/istio/multicluster/kiali-create-remote-cluster-secret.sh

My thinking is we either fix up/enhance that script so it is useful to users in production or (what I think I would prefer, at least that is just initial impressions) is have the operator run in the clusters and you create a Kiali CR to tell the operator "don't install Kiali pod, but do create the roles/rolebindings/service-accounts". That could be easily implemented by a new `deployment.roles_only` setting in the CR - this would tell the operator to not deploy Kiali itself, but do deploy the roles and service account.

At that point, the remote secrets can be created that use the remote SAs (that would be another task to have to figure out how to help the user - they need to create a remote secret that uses the Kiali SAs in the remote clusters).

**Comment by jshaughn** [comment]
I agree that we probably need something to help users do the configuration.  I don't think I like the need to install the operator on the remote clients just to generate some artifacts, it smells like an "agent" requirement.  We already have some users complaining that Kiali is a separate installer from say, Maistra.

I think I'd prefer a script that could be run one time, if necessary, by users on each necessary cluster.  Perhaps it could also be run to gather "status" info for debug, issue reports, etc.

**Comment by jmazzitelli** [comment]
If we want to concentrate on a script, we already have this which utilizes our server helm chart:  https://github.com/kiali/kiali/blob/master/hack/istio/multicluster/kiali-create-remote-cluster-secret.sh

This will create the ClusterRole and related resources on the remote cluster. It will also create the remote secret in the namespace where kiali will be or is installed. Since it uses the server helm chart, it will only create the ClusterRole, there is no option to create individual Roles per accessible namespace - that is a feature of the operator only.

We could modify this script a little. Make the creation of the remote secret optional (so the script only creates the resources on the remote cluster - the role/binding/service account - which is what we are discussing in this issue).

I also want this script to accept "current" for one of the kube context names (so if you are already connected and logged into the remote cluster and it is the current kube context, you can just tell the script use what you are already connected to currently).

I will do a WIP impl and see what it would look like.

**Comment by jmazzitelli** [comment]
The PR is ready for people to review and see if its good enough for people other than kiali devs to use (i.e. should we document it on kiali.io?)

The `--help` provides information on all the options.

`hack/istio/multicluster/kiali-prepare-remote-cluster.sh -h`

See the PR comments for some examples of tests I ran.
